src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -140,15 +147,38 @@ private constructor(
             extensionMessage.isMessageInProgress == false) {
           getCancellationToken().dispose()
         } else {
-          if (lastMessage?.text != null && extensionMessage.chatID != null) {
-            addAssistantResponseToChat(lastMessage.text, lastMessage.displayText)
+
+          if (extensionMessage.chatID != null) {
+            if (prevLastMessage != null) {
+              if (lastMessage?.contextFiles != messages.lastOrNull()?.contextFiles) {
+                val messageId = extensionMessage.messages.count() - 2
```

Comment:
When we assign messageId for a first time it's `id = messages.count()`
Why here we need to substract `1`/`2` and not `0`/`1`?
`extensionMessage` have more messages than our `messages`?

Reply:
`id` is an index in the incoming `extensionMessage.messages`- I just calculate that index

Reply:
I know what `id` is, it's not what I was asking about ;)
To me it just seems to be off by one because you need to subtract `-2` instead of `-1` and I do not fully understand why.

Reply:
`prevLastMessage` has index `extensionMessage.messages.count() - 2`

we are numbering items `0` to `count - 1`

Reply:
Ah, because when we get messages count message is not tagged yet. I see.